gradle-twirl/src/test/java/play/twirl/gradle/AbstractFunctionalTest.java

```diff
@@ -98,14 +98,17 @@ protected BuildResult build(String gradleVersion, String... args) {
   }
 
   static Stream<String> gradleVersions() {
-    // https://docs.gradle.org/current/userguide/scala_plugin.html#sec:configure_zinc_compiler
-    if (getScalaVersion().equals("3")) { // Gradle 7.5+
-      return Stream.of("7.6.2", "8.0.2", "8.7");
-    }
     // https://docs.gradle.org/current/userguide/compatibility.html
+    if (JavaVersion.current().compareTo(JavaVersion.VERSION_21) >= 0) { // Gradle 8.4+
+      return Stream.of("8.7");
+    }
     if (JavaVersion.current().compareTo(JavaVersion.VERSION_17) >= 0) { // Gradle 7.3+
       return Stream.of("7.6.2", "8.0.2", "8.7");
     }
+    // https://docs.gradle.org/current/userguide/scala_plugin.html#sec:configure_zinc_compiler
+    if (getScalaVersion().equals("3")) { // Gradle 7.5+
+      return Stream.of("7.6.2", "8.0.2", "8.7");
+    }
```

Comment:
@ihostage I had to adjust this section, because Java 21 is supported by gradle 8.4+, so I moved the java checks to the top. Should be good.